machineconfiguration/v1/types.go

```diff
@@ -167,6 +185,32 @@ type ControllerConfigStatus struct {
 	// conditions represents the latest available observations of current state.
 	// +optional
 	Conditions []ControllerConfigStatusCondition `json:"conditions" protobuf:"bytes,2,rep,name=conditions"`
+	// controllerCertificates represents the latest available observations of the automatically rotating certificates in the MCO.
+	// +optional
+	ControllerCertificates []ControllerCertificate `json:"controllerCertificates"`
+}
+
+// ControllerCertificate contains info about a specific cert.
+type ControllerCertificate struct {
+	// subject is the cert subject
+	// +kubebuilder:validation:Required
+	Subject string `json:"subject"`
+
+	// signer is the  cert Issuer
+	// +kubebuilder:validation:Required
+	Signer string `json:"signer"`
+
+	// notBefore is the lower boundary for validity
+	// +kubebuilder:validation:Required
+	NotBefore string `json:"notBefore"`
+
+	// notAfter is the upper boundary for validity
+	// +kubebuilder:validation:Required
+	NotAfter string `json:"notAfter"`
```

Comment:
Not sure that this matters, I added these and for sanity of output string just worked better. I can change if we need to

Reply:
Is this status or spec?

Typically timestamps in kube are all `*metav1.Time` since it has special marshal/unmarshal rules and a standard representation of the time, not sure it's hugely important but had I been doing an API review on this, I would have suggested to be consistent with the other APIs

Reply:
Yeah I think that would be preferable if these fields were introduced this release

Reply:
I've changed these time fields to *metav1.Time, I left it as a separate commit but I can squash it out if need be

Reply:
It sounds like changing this is tricky for backwards compatibility?  Since controllerconfig is an implementation detail today, I'm not sure we should block on it.